bugsnag-plugin-android-ndk/src/main/jni/bugsnag.c

```diff
@@ -81,16 +81,30 @@ void bsg_populate_notify_stacktrace(JNIEnv *env, bugsnag_stackframe *stacktrace,
       goto exit;
     }
 
+    // populate filename
     jstring filename = bsg_safe_new_string_utf(env, frame.filename);
-    jstring method;
+    if (filename == NULL) {
+      goto exit;
+    }
+
+    // populate method
+    jstring method = NULL;
     if (strlen(frame.method) == 0) {
-      char *frame_address = calloc(1, sizeof(char) * 32);
-      sprintf(frame_address, "0x%lx", (unsigned long)frame.frame_address);
-      method = bsg_safe_new_string_utf(env, frame_address);
-      free(frame_address);
+      size_t frame_len = sizeof(char) * 32;
+      char *frame_address = calloc(1, frame_len);
```

Comment:
Since this is only 32 bytes, why not just allocate it on the stack?

Reply:
Good idea - I've tweaked the code to do this.